js/coinw.js

```diff
@@ -0,0 +1,588 @@
+        //               "last":"0.008",
+        //               "highestBid":"0.007",
+        //               "id":39,
+        //               "isFrozen":0
+        //               "baseVolume":"0.0",
+        //               "lowestAsk":"0.008",
+        //               "low24hr":"0.0"
+        //           }
+        //       },
+        //       "msg":"SUCCESS"
+        //   }
+        //
+        const data = this.safeValue (response, 'data', {});
+        const tickers = this.parseTickers (data, symbols);
+        const result = this.indexBy (tickers, 'symbol');
+        return this.filterByArray (result, 'symbol', symbols);
+    }
+
+    parseTrade (trade, market = undefined) {
+        //
+        //    {
+        //        amount: '0.6463',
+        //        id: '76700619',
+        //        price: '16956.08',
+        //        time: '2022-12-02 23:14:34',
+        //        total: '10958.714504',
+        //        type: 'SELL'
+        //    }
+        //
+        const datetime = this.safeString (trade, 'time');
+        const orderId = this.safeString (trade, 'id');
+        return this.safeTrade ({
+            'info': trade,
+            'id': orderId,
+            'order': orderId,
+            'timestamp': this.parse8601 (datetime),
+            'datetime': datetime,
+            'symbol': market['symbol'],
+            'type': undefined,
+            'takerOrMaker': undefined,
+            'side': this.safeStringLower (trade, 'type'),
+            'price': this.safeString (trade, 'price'),
+            'amount': this.safeString (trade, 'amount'),
+            'cost': this.safeString (trade, 'total'),
+            'fee': {
+                'cost': undefined,
+                'currency': undefined,
+            },
+        }, market);
+    }
+
+    async fetchTrades (symbol, since = undefined, limit = undefined, params = {}) {
+        /**
+         * @method
+         * @name coinw#fetchTrades
+         * @description get the list of most recent trades for a particular symbol
+         * @param {string} symbol unified symbol of the market to fetch trades for
+         * @param {int|undefined} since timestamp in ms of the earliest trade to fetch
+         * @param {int|undefined} limit the maximum number of trades structures to retrieve, default = 100, max = 1000
+         * @param {object} params extra parameters specific to the coinw api endpoint
+         * @param {int|undefined} params.end the earliest time in ms to fetch trades for
+         * @returns {[object]} a list of [trade structures]{@link https://docs.ccxt.com/en/latest/manual.html?#public-trades}
+         */
+        await this.loadMarkets ();
+        const market = this.market (symbol);
+        const end = this.safeInteger (params, 'end');
+        const request = {
+            'symbol': market['id'],
+        };
+        if (since !== undefined) {
+            request['start'] = since;
+        }
+        if (end !== undefined) {
```

Comment:
we don't need to remove it from `params` to place it inside `request` because in the end we always do `this.extend(request,params)`